src/libcrun/criu.c

```diff
@@ -367,9 +372,19 @@ libcrun_container_checkpoint_linux_criu (libcrun_container_status_t *status, lib
           if (UNLIKELY (ret < 0))
             return crun_make_error (err, errno, "unable to stat(): `%s`", def->linux->namespaces[i]->path);
 
-          xasprintf (&external, "net[%ld]:extRootNetNS", statbuf.st_ino);
+          xasprintf (&external, "net[%ld]:" CRIU_EXT_NETNS, statbuf.st_ino);
+          criu_add_external (external);
+        }
+
+      if (value == CLONE_NEWPID && def->linux->namespaces[i]->path != NULL)
+        {
+          struct stat statbuf;
+          ret = stat (def->linux->namespaces[i]->path, &statbuf);
+          if (UNLIKELY (ret < 0))
+            return crun_make_error (err, errno, "unable to stat(): `%s`", def->linux->namespaces[i]->path);
+
+          xasprintf (&external, "pid[%ld]:" CRIU_EXT_PIDNS, statbuf.st_ino);
```

Comment:
we reuse external here and end up leaking it.

I think it is better to just define `external` in the loop scope:

```
diff --git a/src/libcrun/criu.c b/src/libcrun/criu.c
index 3582ec1..262e8f1 100644
--- a/src/libcrun/criu.c
+++ b/src/libcrun/criu.c
@@ -217,7 +217,6 @@ libcrun_container_checkpoint_linux_criu (libcrun_container_status_t *status, lib
   cleanup_free char *descriptors_path = NULL;
   cleanup_free char *freezer_path = NULL;
   cleanup_close int descriptors_fd = -1;
-  cleanup_free char *external = NULL;
   cleanup_free char *path = NULL;
   cleanup_close int image_fd = -1;
   cleanup_close int work_fd = -1;
@@ -361,6 +360,7 @@ libcrun_container_checkpoint_linux_criu (libcrun_container_status_t *status, lib
 
   for (i = 0; i < def->linux->namespaces_len; i++)
     {
+      cleanup_free char *external = NULL;
       int value = libcrun_find_namespace (def->linux->namespaces[i]->type);
       if (UNLIKELY (value < 0))
         return crun_make_error (err, 0, "invalid namespace type: `%s`", def->linux->namespaces[i]->type);
```

Reply:
Thanks, I've updated the pull request.